### PR TITLE
Books

### PR DESCRIPTION
**Pull request by felix-waller-softwire**
List, show, create, edit, delete books

### CODE REVIEW

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -1,4 +1,33 @@
...
+  def show
+    @book = Book.find(params[:id])
+  end
+
+  def new
+    @book = Book.new
+  end
+
+  def create
+    @book = Book.create(params.require(:book).permit(:title, :author, :isbn))
```
Comment (liliannaszabo) [inline app/controllers/books_controller.rb]: Wouldn't it be nicer to have this a variable so if we need to change the columns we only need to change it in one place?

File: app/views/books/_book_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<%= form_with model: @book do |form| %>
+  <%= form.label :title, 'Title' %>
+  <%= form.text_field :title %>
+  <%= form.label :author, 'Author' %>
+  <%= form.text_field :author %>
+  <%= form.label :isbn, 'ISBN' %>
+  <%= form.text_field :isbn %>
+  <%= form.submit %>
+<% end %>
```
Comment (liliannaszabo) [inline app/views/books/_book_form.html.erb:9]: This will let empty fields be submitted. Was that the intended behaviour?

Comment (dbarnettS) [reply app/views/books/_book_form.html.erb:9]: Why would it @liliannaszabo ?

File: app/views/books/_book_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<%= form_with model: @book do |form| %>
```
Comment (dbarnettS) [inline app/views/books/_book_form.html.erb:1]: Good use of partials!

File: app/views/books/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+<%= link_to 'Books', books_path %>
+<%= link_to 'Edit', edit_book_path(@book) %>
+<%= link_to 'Delete', @book, data: {
+	turbo_method: :delete,
+	turbo_confirm: 'Are you sure?'
+}%>
+
+<h2>copies</h2>
+<ul>
```
Comment (dbarnettS) [inline app/views/books/show.html.erb:11]: A partial can be used here.

File: config/routes.rb [hunk 5]
```diff
@@ -1,7 +1,3 @@
 Rails.application.routes.draw do
-  get 'books/index'
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  resources :books
```
Comment (dbarnettS) [inline config/routes.rb:2]: Nice use of resources!

**Review by dbarnettS — APPROVED** [review]
Looks very good with nice use of ruby on rails syntax e.g. partials. Although there are more cases for partials plus models (Book and Copy) could benefit from validation.

File: app/views/books/_book_form.html.erb (hunk 2 above)
Comment (felix-waller-softwire) [reply app/views/books/_book_form.html.erb:9]: I don't have validations on the model and this form doesn't enforce any client-side validations

File: app/views/books/_book_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+  <%= form.label :title, 'Title' %>
+  <%= form.text_field :title %>
+  <%= form.label :author, 'Author' %>
+  <%= form.text_field :author %>
+  <%= form.label :isbn, 'ISBN' %>
+  <%= form.text_field :isbn %>
+  <%= form.submit %>
```
Comment (8kristy) [inline app/views/books/_book_form.html.erb:8]: Just improves readability a bit :) 
```suggestion
  <%= form.label :title, 'Title' %>
  <%= form.text_field :title %>

  <%= form.label :author, 'Author' %>
  <%= form.text_field :author %>

  <%= form.label :isbn, 'ISBN' %>
  <%= form.text_field :isbn %>

  <%= form.submit %>
```

**Review by 8kristy — COMMENTED** [review]
Nice! I think it would be good to think about and add some validation and think about various edge cases, and then add some test for all of them
